File: Day4/day4.ts
import fs from 'fs';
const path = require('path');

const filePath = path.join(__dirname, 'data.txt');
const file = fs.readFileSync(filePath, 'utf-8');

const inputArray = file.split('\n')

//iterate over inputArray
let item: Array<string>
let first: any;
let second: any; 
let range1Start: any;
let range1End: any;
let range2Start: any;
let range2End: any;
let sum: number = 0; 
for(let i = 0; i < inputArray.length; i++){
   item = inputArray[i].split(',');
   const [one, two] = item;
   first = one.split('-');
   second = two.split('-');
   range1Start = Number(first[0]);
   range1End = Number(first[1]);
   range2Start = Number(second[0]);
   range2End = Number(second[1]);
    if ((range1Start >= range2Start && range1End <= range2End) || (range2Start >= range1Start && range1End >= range2End)){
        sum +=1;
    }
}
console.log(sum)

File: Day4/day4.py
with open('/Users/johnwroge/advent_of_code/Day4/data.txt', 'r') as file:
    contents = file.read().splitlines()
    sum = 0
    # for i in range(len(contents)):
    #     info = contents[i]
    #     item = info.split(",")
    #     one, two = item
    #     first = one.split('-')
    #     second = two.split('-')
    #     range1Start = int(first[0])
    #     range1End = int(first[1])
    #     range2Start = int(second[0])
    #     range2End = int(second[1])
    #     if (range1Start >= range2Start and range1End <= range2End) or (range2Start >= range1Start and range1End >= range2End):
    #         sum +=1
    # for i in range(len(contents)):
    #     info = contents[i]
    #     item = info.split(",")
    #     one, two = item
    #     first = one.split('-')
    #     second = two.split('-')
    #     result = []
    #     for number in first:
    #         digits = [int(digit) for digit in number]
    #         result.extend(digits)
    #     print(result)


    #     result2 = []
    #     for number in second:
    #         digits2 = [int(digit) for digit in number]
    #         result2.extend(digits)
    for i in range(len(contents)):
        info = contents[i]
        item = info.split(",")
        one, two = item
        first = one.split('-')
        second = two.split('-')
        range1Start = int(first[0])
        range1End = int(first[1])
        range2Start = int(second[0])
        range2End = int(second[1])
        # (2-4,6-8 and 2-3,4-5)
        if ((range1Start <= range2Start) and (range1End >= range2Start)) or ((range2Start <= range1Start) and (range2End >= range1Start)):
            sum += 1
#correct 905       
print(sum)